web/packages/teleterm/src/ui/DocumentCluster/resourcesContext.tsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EventEmitter } from 'events';
```

Comment:
[`useEventEmitter`](https://ahooks.js.org/hooks/use-event-emitter/) has a nice API and automatically cleans up the listener. However, I didn't want to introduce a fourth implementation of an event emitter (built-in Node.js version, the events package version, the Emittery package version).

Also, if we really wanted to, we could add another custom hook which basically does `onResourcesRefreshRequest` along with `useEffect` and returns the cleanup function. But since this is called in only a single place, clear code is more valuable than ergonomics.

Reply:
I agree that would be too many event emitters :)